TLM/TLM/UI/TrafficManagerTool.cs

```diff
@@ -939,12 +939,56 @@ public class TrafficManagerTool
                     }
                 }
 
-                return HoveredNodeId != 0 || HoveredSegmentId != 0;
+                if (HoveredNodeId != 0) {
+                    HoveredSegmentId = GetHoveredSegmentFromNode();
+                }
             }
 
-            return false; // mouseRayValid=false here
+            return HoveredNodeId != 0 || HoveredSegmentId != 0;
         }
 
+        /// <summary>
+        /// returns the node segment that is closest to the mouse pointer based on angle.
+        /// </summary>
+        internal ushort GetHoveredSegmentFromNode() {
+            bool considerSegmentLenght = true;
```

Comment:
Is there a better place for this hard coded configuration option?

Reply:
a constant somewhere in the same a class, or `Constants.cs` module (we have one or a few)

Reply:
thanks